src/NHibernate/Linq/Visitors/PartialEvaluationResult.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections.Generic;
+using System.Linq.Expressions;
+
+namespace NHibernate.Linq.Visitors
+{
+	/// <summary>
+	/// The result of <see cref="NhRelinqQueryParser.PartialEvaluate"/> method.
+	/// </summary>
+	public class PartialEvaluationResult
```

Comment:
I created a separate class in order to be able to add additional data in the future.

Reply:
Maybe `QueryMode` should be added here?

Reply:
Or maybe not... I see it's not very convenient..

Reply:
> Maybe QueryMode should be added here?

If we add it here, then we would need to add it as internal otherwise would be confusing for users as it is not populated by the `PreTransform` method. But if we add it as internal, then users (also in our tests) cannot simulate what nhibernate does as `QueryMode` would be set internally by the `NhLinqExpression`. That is why I added it as an additional parameter. Also, with #2079 `ExpressionParameterVisitor` will be obsolete.

Reply:
Well `PartialEvaluate` can take `QueryMode` as additional parameter... But OK - maybe it's not worth it.

> Also, with #2079 `ExpressionParameterVisitor` will be obsolete.

Could you please clarify how it correlates with this PR. Will you need to modify `ExpressionKeyVisitor` or what's used instead in similar way as you modified `ExpressionParameterVisitor`?

Reply:
> Well PartialEvaluate can take QueryMode as additional parameter

That is another option which I think it is even better. By passing the information to the `PreTransform` method we could skip the variable detection logic for DML queries and avoid passing the `QueryMode` to `ExpressionParameterVisitor` as `QueryVariables` dictionary will be empty. With the last commit the `QueryMode` is passed into `PreTransform` method instead.

> Will you need to modify ExpressionKeyVisitor or what's used instead in similar way as you modified ExpressionParameterVisitor?

The logic of `ExpressionParameterVisitor` that detects whether an expression is a parameter or not, was moved into `ExpressionKeyVisitor`, other stuff like removing `MappedAs` calls and detecting parameter types was moved in the translation phase. So yes, the `ExpressionKeyVisitor` will need to be modified similar than `ExpressionParameterVisitor` is modified here.